python/cudf/cudf/_libxx/parquet.pyx

```diff
@@ -35,6 +40,40 @@ from cudf._libxx.io.utils cimport (
 cimport cudf._libxx.cpp.types as cudf_types
 cimport cudf._libxx.cpp.io.types as cudf_io_types
 
+cdef class BufferArrayFromVector:
+    cdef unsigned length
+    cdef unique_ptr[vector[uint8_t]] in_vec
+
+    # these two things declare part of the buffer interface
+    cdef Py_ssize_t shape[1]
+    cdef Py_ssize_t strides[1]
+
+    cdef set_ptr(self, unique_ptr[vector[uint8_t]] in_vec):
+        self.in_vec = move(in_vec)
+        self.length = dereference(self.in_vec).size()
```

Comment:
Per the [cython docs](https://cython.readthedocs.io/en/latest/src/userguide/extension_types.html#existing-pointers-instantiation), we can initialize a new `BufferArrayFromVector` instance via a static method like this. 

This pattern also matches what we do with [`Table.from_unique_ptr()`](https://github.com/rapidsai/cudf/blob/branch-0.14/python/cudf/cudf/_libxx/table.pyx#L72-L77):
```suggestion
    @staticmethod
    cdef BufferArrayFromVector from_unique_ptr(unique_ptr[vector[uint8_t]] in_vec):
      buf = BufferArrayFromVector()
      buf.in_vec = move(in_vec)
      buf.length = dereference(self.in_vec).size()
      return buf
```

Reply:
Trying this out now.

Reply:
@trxcllnt this seems to cause a segfault right now, most likely something going wrong with the pointers. I am investigating and will update here once I understand what's happening.

Reply:
I think it needs to be this instead:
```
buf.length = dereference(buf.in_vec).size()
```

Reply:
Yeah good catch. Always dicey writing code in the github UI :-)